Sources/Client.swift

```diff
@@ -314,9 +297,19 @@ public class CentrifugeClient {
             do {
                 let paramsData = try params.serializedData()
                 let command = strongSelf.newCommand(method: .ping, params: paramsData)
-                let _ = try strongSelf.sendCommand(command: command)
-            } catch CentrifugeError.timeout {
-                strongSelf.close(reason: "no ping", reconnect: true)
+                strongSelf.sendCommand(command: command, completion: { [weak self] res, error in
+                    guard let strongSelf = self else { return }
+                    if let err = error {
+                        switch err {
+                        case CentrifugeError.timeout:
+                            strongSelf.close(reason: "no ping", reconnect: true)
```

Comment:
In this case would be better
if let err = error, err == .timeout {do something}

Reply:
tried this - does not compile unfortunately:
```
Binary operator '==' cannot be applied to operands of type 'Error' and 'CentrifugeError'
```